src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -90,12 +90,37 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
             return;
         }
 
+        self::inline($classes, $cache, $declared);
+
+        if ($autoReload) {
+            // save the resources
+            self::writeCacheFile($metadata, serialize(array($files, $classes)));
```

Comment:
this is broken, because ``$files`` is modified into ``inline``.

Savign resources should stay at the same place than saving the main file btw.